roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/entity/RoundaboutCarTest.java

```diff
@@ -126,42 +127,84 @@ public void getNextSection_currentSectionNotEqualDestination() {
+        ICar iCar = new RoundaboutCar(car, 10, driverBehaviourMock, route);
 
         assertEquals(iCar.getCurrentSection(), currentSectionMock);
         assertNotEquals(iCar.getCurrentSection(), destinationMock);
     }
 
     @Test
     public void getTimeToTraverseCurrentSection_isExpectedTime() {
-        RoundaboutSimulationModel model = getPrepareModel();
+        RoundaboutSimulationModel model = getPreparedModel();
         Street section = new StreetSection(10.0, model, "", false);
-        IDriverBehaviour driverBehaviour = new DriverBehaviour(5.0, 2.0, 2.0, 1.5);
+        IDriverBehaviour driverBehaviour = new DriverBehaviour(5.0, 2.0, 2.0, 1.5, 1);
 
         IRoute route = new Route();
         route.addSection(section);
 
         Car car = new Car(model, "", false);
-        ICar iCar = new RoundaboutCar(car,2, driverBehaviour, route);
+        ICar iCar = new RoundaboutCar(car, 2, driverBehaviour, route);
 
         Assert.assertEquals(2.0, iCar.getTimeToTraverseCurrentSection(), 0.0);
     }
 
     @Test
-    public void getTimeToTraverseSection_carIsNotOnThisSection(){
-        RoundaboutSimulationModel model = getPrepareModel();
+    public void getTimeToTraverseSection_carIsNotOnThisSection() {
+        RoundaboutSimulationModel model = getPreparedModel();
         Street currentSection = new StreetSection(10.0, model, "", false);
         Street otherSection = new StreetSection(20.0, model, "", false);
-        IDriverBehaviour driverBehaviour = new DriverBehaviour(5.0, 2.0, 2.0, 1.5);
+        IDriverBehaviour driverBehaviour = new DriverBehaviour(5.0, 2.0, 2.0, 1.5, 1);
 
         IRoute route = new Route();
         route.addSection(currentSection);
 
         Car car = new Car(model, "", false);
-        ICar iCar = new RoundaboutCar(car,2, driverBehaviour, route);
+        ICar iCar = new RoundaboutCar(car, 2, driverBehaviour, route);
 
         Assert.assertEquals(4, iCar.getTimeToTraverseSection(otherSection), 0.0);
 
     }
+
+    @Test
+    public void getTransitionTime_noJam() {
+        RoundaboutSimulationModel modelMock = mock(RoundaboutSimulationModel.class);
+        when(modelMock.getCurrentTime()).thenReturn(5.0);
+
+        Car oldCarMock = mock(Car.class);
+        when(oldCarMock.getModel()).thenReturn(modelMock);
+
+        RoundaboutCar carMock = mock(RoundaboutCar.class);
+        when(carMock.getTransitionTime()).thenCallRealMethod();
+        when(carMock.getOldImplementationCar()).thenReturn(oldCarMock);
+
+        when(carMock.getTimeToTraverseCurrentSection()).thenReturn(8.0);
+        when(carMock.getLastUpdateTime()).thenReturn(1.0);
+
+        assertEquals(8.0, carMock.getTransitionTime(), 0.0);
+    }
+
+    @Test
+    public void getTransitionTime_withJam() {
+        RoundaboutSimulationModel modelMock = mock(RoundaboutSimulationModel.class);
+        when(modelMock.getCurrentTime()).thenReturn(5.0);
+        when(modelMock.getStandardCarAccelerationTime()).thenReturn(2.0);
```

Comment:
why is the standard acceleration time needed when there is a jam? (i dont understand the connection between them)